lib/pure/terminal.nim

```diff
@@ -448,9 +488,18 @@ type
 
 when not defined(windows):
   var
-    # XXX: These better be thread-local
-    gFG = 0
-    gBG = 0
+    gFG {.threadvar.}: int
+    gBG {.threadvar.}: int
```

Comment:
There is far too many global thread variables in this module (and this isn't your fault). But these should all be put into a single ``State`` object, eventually at least.